pr-dhl-woocommerce/includes/class-pr-dhl-wc-notice-soap-deprecation.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * SOAP API Deprecation Admin Notice.
+ *
+ * @package dhl-for-woocommerce
+ */
+
+
+use PR\DHL\Utils\API_Utils;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+if ( ! class_exists( 'PR_DHL_WC_Notice_SOAP_Deprecation' ) ) :
+
+	class PR_DHL_WC_Notice_SOAP_Deprecation {
+
+		/**
+		 * Register hooks.
+		 */
+		public static function init() {
+			// Run late enough that WooCommerce + settings are available.
+			add_action( 'admin_notices', array( __CLASS__, 'render' ) );
+		}
+
+		/**
+		 * Decide whether to show the notice.
+		 *
+		 * @return bool
+		 */
+		protected static function should_show() {
+			if ( ! is_admin() ) {
+				return false;
+			}
```

Comment:
Since `render()` is only hooked into `admin_notices`, which runs `exclusively in the admin context`, there is no need to ad this check here